### PR TITLE
Safari bits offer options

### PR DESCRIPTION
**Pull request by zaxy78**
Fix for #785 + unit tests (legacy createOffer shim fix)

### CODE REVIEW

File: src/js/safari/safari_shim.js [hunk 1]
```diff
@@ -259,6 +259,12 @@ module.exports = {
           return transceiver.sender.track &&
               transceiver.sender.track.kind === 'audio';
         });
+        var isBit = offerOptions.offerToReceiveAudio;
```
Comment (KaptenJansson) [inline src/js/safari/safari_shim.js]: Since you are using it in two places and its quite a few lines, perhaps pull it out to a helper function instead?

**Review by KaptenJansson — CHANGES_REQUESTED** [review]

File: src/js/safari/safari_shim.js [hunk 2]
```diff
@@ -278,6 +279,7 @@ module.exports = {
           pc.addTransceiver('audio');
         }
 
+        pc.offerToReceiveVideo = !!pc.offerToReceiveVideo; // support bit value
```
Comment (zaxy78) [inline src/js/safari/safari_shim.js]: found a much simpler way to fix this. 
but gave up the 'deprecation' message I've put there.

File: src/js/safari/safari_shim.js [hunk 3]
```diff
@@ -278,6 +280,8 @@ module.exports = {
           pc.addTransceiver('audio');
         }
 
+        // support bit value
+        offerOptions.offerToReceiveVideo = !!offerOptions.offerToReceiveVideo;
```
Comment (zaxy78) [inline src/js/safari/safari_shim.js]: @KaptenJansson - I found a much simpler way to fix this.
but gave up the 'deprecation' message I've put there.

Comment (fippo) [reply src/js/safari/safari_shim.js]: if offerToReceiveVideo is unset the !! will make it false which may change the direction. The simplest way might be to check for "this isn't undefined"

**Comment by zaxy78** [comment]
@KaptenJansson looks better?

File: src/js/safari/safari_shim.js [hunk 4]
```diff
@@ -278,6 +282,11 @@ module.exports = {
           pc.addTransceiver('audio');
         }
 
+
+        if (offerOptions.offerToReceiveVideo !== undefined) {
+          // support bit values
+          offerOptions.offerToReceiveVideo = !!offerOptions.offerToReceiveVideo;
+        }
```
Comment (zaxy78) [inline src/js/safari/safari_shim.js]: @KaptenJansson looks better? I think It's ready for merge

File: src/js/safari/safari_shim.js [hunk 5]
```diff
@@ -255,6 +255,10 @@ module.exports = {
     window.RTCPeerConnection.prototype.createOffer = function(offerOptions) {
       var pc = this;
       if (offerOptions) {
+        if (offerOptions.offerToReceiveAudio !== undefined) {
```
Comment (KaptenJansson) [inline src/js/safari/safari_shim.js]: I would prefer typeof instead even though its not used everywhere in this repo.

`if (typeof offerOptions.offerToReceiveVideo !== 'undefined') {
.... }`

**Review by KaptenJansson — CHANGES_REQUESTED** [review]

File: src/js/safari/safari_shim.js (hunk 5 above)
Comment (zaxy78) [reply src/js/safari/safari_shim.js]: @KaptenJansson  done

**Comment by zaxy78** [comment]
Fixed and Ready to merge.

Also, @KaptenJansson, when do you plan to have the next release?

I think after #783, we can add "Support for iOS WKWebView" and make a new release

**Comment by zaxy78** [comment]
@KaptenJansson, thanks for the release publish. any reason not to merge this?
I fixed your comments..

**Review by KaptenJansson — APPROVED** [review]

**Pull request merged**